### ISSUE TITLE
Does not print without any errors (cups-filters-1.27.1)

### ISSUE THREAD

**Issue by a-baturin** [OpenPrinting/cups-filters#208]
Hi
Fedora 31, kernel 5.5.5-200
Ricoh SP C250SF on Ethernet. built-id Ricoh 440 and PPD from http://www.openprinting.org/printer/Ricoh/Ricoh-SP_C250SF
I upgraded cups-filters-1.22.5-4 to 1.27.1-1 and no printing anymore.
I see messages "printing completed" but job was not really sent to printer.
Downgraded back -> printing fine.
Tried with 2 PCs
[troubleshoot.txt](https://github.com/OpenPrinting/cups-filters/files/4243500/troubleshoot.txt)

**Comment by OdyX** [comment]
This is likely #206

**Comment by a-baturin** [comment]
Only if test page for printer also PostScript.
I can't print even test page from printer settings

**Comment by tillkamppeter** [comment]
I got a fix for Issue #206 now and applied it, Please re-test with the current snapshot of cups-filters.

**Comment by a-baturin** [comment]
Unfortunately still not ok.
Now it wakes printer up, but doesn't print

**Comment by tillkamppeter** [comment]
So we need the CUPS error_log from you, in debug mode. If you have CUPS 2.3.0 or older, run
```
cupsctl --debug-logging
```
to activate debug mode. Do not use `cupsctl` with CUPS 2.3.1 (if the package's ChangeLog does not tell that the problem is fixed). In this case stop the CUPS daemon, edit `/etc/cups/cupsd.conf` to have a line
```
LogLevel debug
```
and start the CUPS daemon again.
Clean up with
```
cancel -a
```
and print another job of the kind which does not print. Wait for it to finish or at least long enough until the printing attempt has happened (printer woken up from sleep for example).
Now take the file
```
/var/log/cups/error_log
```
and attach it here. You need to rename the file adding a `.txt` extension so that it gets accepted as attachment.

**Comment by a-baturin** [comment]
Tried this (thanks for detailed description btw)
Unfortunately error_log is empty after it

**Comment by rickrich** [comment]
You guys should remove 1.27.1 because it will not work with my set of appox. 50 filters.
http://foo2zjs.rkkda.com/

**Comment by tillkamppeter** [comment]
@rickrich, could you please test with the current GIT? If it works correctly I will put out 1.27.2 immediately.

**Comment by rickrich** [comment]
I tested it on Fedora 31 in a Virtual Box because my main machine runs Fedora 30.

cups-filters-1.27.1-2.fc31.x86_64.rpm
cups-filters-libs-1.27.1-2.fc31.x86_64.rpm

**Comment by tillkamppeter** [comment]
I have now tested both PDF and PostScript files and in each format files with 7 pages, 1 page, and 0 pages. I have also tested the CUPS test page `/usr/share/cups/data/testprint`. I have tested all this with the Ricoh PPD of the initial posting and with the foo2zjs-based PPD file for the Samsung CLX-3160. Using the current state of the GIT repository all these tests lead to correct results for me. 

I have done this test on Ubuntu 20.04 with all the printer-driver-foo2zjs package of Ubuntu installed. The Ricoh PPD I have doesnloaded from the OpenPrinting web site.

Seems that the last commits have fixed this bug. So I am closing it here and releasing 1.27.2.
